shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java

```diff
@@ -61,20 +60,88 @@
           .argName("replicas")
           .desc("Number of block replicas of each loaded file, default: " + DEFAULT_REPLICATION)
           .build();
-  private static final int DEFAULT_PARALLELISM = 10;
-  private static final Option PARALLELISM_OPTION =
+
+  private static final int DEFAULT_ACTIVE_JOBS = 1000;
+  private static final Option ACTIVE_JOBS_OPTION =
       Option.builder()
-          .longOpt("parallelism")
+          .longOpt("active_jobs")
           .required(false)
           .hasArg(true)
-          .argName("# concurrent operations")
           .numberOfArgs(1)
-          .desc("Number of concurrent load operations, default: " + DEFAULT_PARALLELISM)
+          .type(Number.class)
+          .argName("jobs")
+          .desc("Maximum number of active outgoing jobs, default: " + DEFAULT_ACTIVE_JOBS)
           .build();
 
-  private ThreadPoolExecutor mDistributedLoadServicePool;
-  private CompletionService<AlluxioURI> mDistributedLoadService;
-  private ArrayList<Future<AlluxioURI>> mFutures = new ArrayList<>();
+  private class JobAttempt {
+    private final JobConfig mJobConfig;
+    private final RetryPolicy mRetryPolicy;
+    private final JobMasterClient mClient;
+
+    private Long mJobId;
+
+    private JobAttempt(JobConfig jobConfig, RetryPolicy retryPolicy, ClientContext clientContext) {
+      mJobConfig = jobConfig;
+      mRetryPolicy = retryPolicy;
+      mClient = JobMasterClient.Factory.create(
+          JobMasterClientContext.newBuilder(clientContext).build());
```

Comment:
+1 to pooling the clients, especially since these are mostly sharable (all going to job master), we could just fetch a resource from a pool each time we need to run an RPC.

Reply:
pooling is very annoying here because the clientContext for each one of these is different. It's also no different from what it was before. (Sicne JobGrpcClientUtils... will create a new client every time)